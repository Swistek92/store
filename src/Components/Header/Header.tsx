import React from "react";
import styles from "./styles.module.css";
import { FaHeart, FaShoppingCart, FaSearch } from "react-icons/fa";
import { NavLink } from "react-router-dom";
import LoginModal from "../Modals/Login/Login";
import RegisterModal from "../Modals/Register/Register";
import { useAppDispatch, useAppSelector } from "../../store/store";
import { showLogin, showRegister } from "../../store/features/ModalSlice";
import {
  Container,
  Nav,
  Navbar,
  NavDropdown,
  Offcanvas,
} from "react-bootstrap";
import { Link } from "react-router-dom";
import HeaderSearch from "./HeaderSearch/HeaderSearch";

const Header = () => {
  const dispatch = useAppDispatch();
  const showLoginModal = () => dispatch(showLogin());
  const showRegisterModal = () => dispatch(showRegister());
  const { user, isError, isLogin, errorMessage, isLoading } = useAppSelector(
    (state) => state.user
  );
  console.log(user);
  return (
    <Navbar
      fixed='top'
      bg='dark'
      variant='dark'
      expand={"xl"}
      className={`mb-3 ' ${styles.navbar}`}
    >
      <Container fluid>
        <Navbar.Brand className={styles.brand}>
          <Link to={`/`}>Home</Link>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls={`offcanvasNavbar-expand-xl`} color='' />
        <Navbar.Offcanvas
          id={`offcanvasNavbar-expand-xl`}
          aria-labelledby={`offcanvasNavbarLabel-expand-xl`}
          placement='end'
          className={styles.offcanvas}
        >
          <Offcanvas.Header closeButton>
            <Offcanvas.Title id={`offcanvasNavbarLabel-expand-xl`}>
              Offcanvas
            </Offcanvas.Title>
          </Offcanvas.Header>
          <Offcanvas.Body>
            <Nav className='justify-content-end flex-grow-1 pe-3'>
              {!isLogin && (
                <Nav.Link onClick={showRegisterModal}>Register</Nav.Link>
              )}
              {!isLogin && <Nav.Link onClick={showLoginModal}>Login</Nav.Link>}

              {isLogin && (
                <NavDropdown
                  className={styles.dropdown}
                  title={`user: ${user.name}`}
                  id={`offcanvasNavbarDropdown-expand-xl`}
                >
                  <NavLink to={`/user/`}>User Page</NavLink>
                  <NavLink to={`/user/profile`}>Profile</NavLink>
                  <NavDropdown.Divider />
                  <NavLink to={`/user/logout`}>logout</NavLink>
                </NavDropdown>
              )}
            </Nav>
            <HeaderSearch />
            {/* palce for search */}
          </Offcanvas.Body>
        </Navbar.Offcanvas>
      </Container>
      <LoginModal />
      <RegisterModal />
    </Navbar>
  );
  // return (
  //   <header className={styles.header}>
  //     <div className={styles.container}>
  //       <div className={styles.logo}>
  //         <p>eStore</p>
  //       </div>
  //       <Button variant='primary' onClick={showLoginModal}>
  //         LoginModal
  //       </Button>
  //       <Button variant='primary' onClick={showRegisterModal}>
  //         RegisterModal
  //       </Button>
  //       <div className={styles.inputGroup}>
  //         <select>
  //           <option>All</option>
  //           <option>MEN</option>
  //           <option>Kieds</option>
  //           <option>Woman</option>
  //           <option>Kids</option>
  //         </select>
  //         <input
  //           placeholder='search'
  //           className={styles.searchInput}
  //           type='text'
  //         />
  //         {/* <button className={styles.searchIcon}>
  //           <FaSearch />
  //         </button> */}
  //       </div>
  //       <div className={styles.links}>
  //         <div className={styles.widget}>
  //           <a href='#'>sign in</a>
  //           <span> / </span>
  //           <a href='#'> sign up </a>
  //         </div>
  //         <div>
  //           <FaHeart style={{ color: "red" }} className={styles.icon} />
  //           <FaShoppingCart className={styles.icon} />
  //         </div>
  //       </div>
  //     </div>
  //     <LoginModal />
  //     <RegisterModal />
  //   </header>
  // );
};

export default Header;
