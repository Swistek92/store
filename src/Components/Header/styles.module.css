.header {
  position: fixed;
  z-index: 100;
}

.container {
  display: flex;
  justify-content: space-around;
  align-items: center;
  position: fixed;
  width: 100vw;
  height: fit-content;
  height: 3.5rem;
  background-color: #333333a3;
}

.logo {
  color: white;
  /* display: none; */
}

.searchInput {
  margin-right: 20px;
}

.inputGroup {
  border-radius: 5px;
  height: 1.5rem;
}

.inputGroup select {
  height: 1.5rem;
  background-color: green;
  color: white;
}

.inputGroup input {
  max-width: 30vw;
  height: 1.5rem;
  display: inline-block;
  margin: 0;
  padding: 0;
  text-align: center;
  /* flex-direction: ; */
}

.icon {
  margin-right: 5px;
  color: #00000073;
}
.icon:first-child {
  margin-left: 5px;
}

.searchIcon {
  color: #000;
  background-color: #fff;
  height: 1.5rem;
  margin: 0;
  padding: 0;
  border: none;
}

.links {
  display: flex;
  /* flex-direction: row; */
  /* align-items: center; */
  /* justify-content: space-between; */
}

.widget a {
  text-decoration: none;
  color: #fff;
  font-size: 10px;
}

@media (min-width: 500px) {
  /* â€¦ */
  .widget a {
    font-size: 15px;
  }
}
