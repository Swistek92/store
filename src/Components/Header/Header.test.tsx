import React from "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Header from "../Header/Header";
import renderWithStore from "../../utils/testing/renderWithStore";
test("snapshot test", () => {
  expect(render(<Header />)).toMatchSnapshot();
});
