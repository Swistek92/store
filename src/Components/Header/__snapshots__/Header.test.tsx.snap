// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="logo"
          >
            <p>
              estore
            </p>
          </div>
          <div
            class="inputGroup"
          >
            <select>
              <option>
                All
              </option>
              <option>
                MEN
              </option>
              <option>
                Kieds
              </option>
              <option>
                Woman
              </option>
              <option>
                Kids
              </option>
            </select>
            <input
              class="searchInput"
              placeholder="search"
              type="text"
            />
          </div>
          <div>
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: red;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              />
            </svg>
            <svg
              class="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
              />
            </svg>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="logo"
        >
          <p>
            estore
          </p>
        </div>
        <div
          class="inputGroup"
        >
          <select>
            <option>
              All
            </option>
            <option>
              MEN
            </option>
            <option>
              Kieds
            </option>
            <option>
              Woman
            </option>
            <option>
              Kids
            </option>
          </select>
          <input
            class="searchInput"
            placeholder="search"
            type="text"
          />
        </div>
        <div>
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: red;"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
            />
          </svg>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
